src/Datadog.Trace.ClrProfiler.Managed/Integrations/Caching/InterceptedMethodAccess.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+
+namespace Datadog.Trace.ClrProfiler.Integrations
+{
+    /// <summary>
+    /// Wrapper class for retrieving instrumented methods and caching them.
+    /// Necessary because our profiling implementation treats all non <see cref="ValueType"/> types as equal for method signature purposes.
+    /// Uses the actual types of arguments from the calling method to differentiate between signatures and caches them.
+    /// </summary>
+    /// <typeparam name="TDelegate">The type of delegate being intercepted.</typeparam>
+    internal class InterceptedMethodAccess<TDelegate>
```

Comment:
I didn't have any issue with `Cache`, but no strong preference here, whatever terminology makes the most sense.